examples/utils/example_node.rs

```diff
@@ -177,20 +192,62 @@ impl ExampleNode {
         }
     }
 
-    // While handling churn messages, we first "action" it ourselves and then
-    // send the corresponding refresh messages out to our close group.
     fn handle_node_added(&mut self, name: XorName) {
-        // TODO: Use the given routing table instead of repeatedly querying the routing node.
         self.send_refresh(MessageId::from_added_node(name));
     }
 
-    fn handle_node_lost(&mut self, name: XorName) {
-        // TODO: Use the given routing table instead of repeatedly querying the routing node.
-        self.send_refresh(MessageId::from_lost_node(name));
+    fn handle_split(&mut self, prefix: Prefix<XorName>) {
+        let id = MessageId::from_added_node(prefix.lower_bound());
+        let mut deleted_clients = Vec::new();
+        for (client_name, stored) in &self.client_accounts {
+            if prefix.matches(client_name) {
+                let refresh_content = RefreshContent::Client {
+                    client_name: *client_name,
+                    data: *stored,
+                };
+
+                let content = unwrap!(serialise(&refresh_content));
+
+                unwrap!(self.node
+                    .send_refresh_request(Authority::ClientManager(*client_name),
+                                          Authority::ClientManager(*client_name),
+                                          content,
+                                          id));
+            } else {
+                deleted_clients.push(*client_name);
+            }
+        }
+        for client in &deleted_clients {
+            let _ = self.client_accounts.remove(client);
+        }
+
+        let mut deleted_data = Vec::new();
+        for (data_name, data) in &self.db {
+            if prefix.matches(data_name) {
+                let refresh_content = RefreshContent::NaeManager {
+                    data_name: *data_name,
+                    data: data.clone(),
+                };
+                let content = unwrap!(serialise(&refresh_content));
+                unwrap!(self.node
+                    .send_refresh_request(Authority::NaeManager(*data_name),
+                                          Authority::NaeManager(*data_name),
+                                          content,
+                                          id));
```

Comment:
Is there a reason to send these refresh messages at all? The other nodes in the group should already have exactly the same entries and will drop them on split, too.